File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int total = brown + yellow;

        for (int height = 3; height <= total / height; height++) {
            if (total % height == 0) {
                int width = total / height;

                if (brown == 2 * (width + height) - 4) {
                    return new int[] { width, height };
                }
            }
        }

        return new int[] {};
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    answer = []
    total = brown + yellow
    '''
    total = 12
    w = 4
    h = 3
    yellow = 2 -> (w-2) * (h-2)
    '''
    for h in range(3, total):
        for w in range(h, total // h+1):
            if (w-2)*(h-2) == yellow:
                return [w,h]